src/opa.rs

```diff
@@ -0,0 +1,369 @@
+//! This module offers common access to the [`OpaConfig`] which can be used in operators
+//! to specify a name for a [`k8s_openapi::api::core::v1::ConfigMap`] and a package name
+//! for OPA rules.
+//!
+//! Additionally several methods are provided to build an URL to query the OPA data API.
+//!
+//! # Example
+//! ```rust
+//! use serde::{Deserialize, Serialize};
+//! use stackable_operator::kube::CustomResource;
+//! use stackable_operator::opa::OpaConfig;
+//! use stackable_operator::schemars::{self, JsonSchema};
+//!
+//! #[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+//! #[kube(
+//!     group = "test.stackable.tech",
+//!     version = "v1alpha1",
+//!     kind = "TestCluster",
+//!     plural = "testclusters",
+//!     shortname = "test",
+//!     namespaced,
+//! )]
+//! #[serde(rename_all = "camelCase")]
+//! pub struct TestClusterSpec {
+//!     opa: Option<OpaConfig>    
+//! }
+//!
+//! let cluster: TestCluster = serde_yaml::from_str(
+//!     "
+//!     apiVersion: test.stackable.tech/v1alpha1
+//!     kind: TestCluster
+//!     metadata:
+//!       name: simple-test
+//!     spec:
+//!       opa:
+//!         configMapName: simple-opa
+//!         package: test
+//!     ",
+//!     ).unwrap();
+//!
+//! let opa_config: &OpaConfig = cluster.spec.opa.as_ref().unwrap();
+//!
+//! assert_eq!(opa_config.package_url(&cluster), "v1/data/test".to_string());
+//! assert_eq!(opa_config.full_package_url(&cluster, "http://localhost:8081"), "http://localhost:8081/v1/data/test".to_string());
+//! assert_eq!(opa_config.rule_url(&cluster, Some("myrule")), "v1/data/test/myrule".to_string());
+//! assert_eq!(opa_config.full_rule_url(&cluster, "http://localhost:8081", Some("myrule")), "http://localhost:8081/v1/data/test/myrule".to_string());
+//! ```
+use crate::client::Client;
+use crate::error;
+use crate::error::OperatorResult;
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::ResourceExt;
+use schemars::{self, JsonSchema};
+use serde::{Deserialize, Serialize};
+
+const OPA_API: &str = "v1/data";
+
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OpaConfig {
+    pub config_map_name: String,
+    pub package: Option<String>,
+}
+
+impl OpaConfig {
+    /// Returns the OPA data API url up to the package. If [`OpaConfig`] has
+    /// no `package` set, will default to the cluster `resource` name.
+    ///
+    /// This may be used if the OPA base url is contained in an ENV variable.
+    ///
+    /// # Example
+    ///
+    /// v1/data/<package>
+    ///
+    /// # Arguments
+    /// * `resource`     - The cluster resource.
+    pub fn package_url<T>(&self, resource: &T) -> String
```

Comment:
nit: The name of this function does not refer to the api (data) or version of the api (v1) that is called at all. Not a severe issue, but if at some point in time we also want to support the query api or v2 of the data api we might end up with inconsistent naming. 
Comment goes for all other fns as well I guess..

Reply:
renamed to document_url (which reflects the data api) and added the version enum.